eva_sub_cli/main.py

```diff
@@ -110,6 +114,28 @@ def get_project_and_vcf_fasta_mapping_from_metadata_xlsx(metadata_xlsx, mapping_
     return project_title, vcf_fasta_report_mapping
 
 
+def check_validation_required(tasks, sub_config):
+    # Validation is mandatory so if submit is requested then VALIDATE must have run before or be requested as well
+    if SUBMIT in tasks:
+        if not sub_config.get(READY_FOR_SUBMISSION_TO_EVA, False):
+            return True
+        else:
+            submission_id = sub_config.get(SUB_CLI_CONFIG_KEY_SUBMISSION_ID, None)
+            if submission_id:
+                try:
+                    submission_status = get_submission_status(submission_id)
+                    if submission_status == 'FAILED':
+                        return True
+                    else:
+                        return False
+                except requests.HTTPError as ex:
+                    logger.error(f'Error occurred while getting status of the submission with ID {submission_id}: {ex.response.status_code} {ex.response.text}')
+                    raise
+            else:
+                logger.info(f'submission id not found. This might be the first time user is submitting')
+                return False
+
```

Comment:
I'd add a `return False` at the very end here, just to be explicit

Reply:
updated